src/components/RepoSubmission.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, useState } from 'react'
+
+
+const RepoSubmission: FC = ()  => {
+  
+  const [buttonPlaceHolder, setButtonPlaceHolder] = useState<string>("Submit repo?")
+  const [isFormOpen, setIsFormOpen] = useState<boolean>(false)
+  const [isSubmissionInProcess, setIsSubmissionInProcess] = useState<boolean>(false)
+  const [submitted, setSubmitted] = useState<boolean>(false)
+  const [input, setInput] = useState("")
+
+  
+  
+  const saveDataToDataBase = (url: string):void => {
+    setIsSubmissionInProcess(true)
+
+    // todo: import the summission function here instead
+    setTimeout(() => {
+
+      setIsSubmissionInProcess(false)
+      setSubmitted(true)
+      setButtonPlaceHolder("Close")
+      console.log("Data Submitted")
+
+    }, 2000);
+  }
+
+  const submitButtonHandler = ():void => {
+    if(!isFormOpen && !submitted){
+      setButtonPlaceHolder("Submit now")
+      return setIsFormOpen(true) 
+    }
+
+    if(isFormOpen && !submitted){
+      saveDataToDataBase(input)
```

Comment:
name of this function can be saveToDataBase

Reply:
Renamed to "saveToDataBase"